sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+        return self._boundary_conditions['moment']
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+        return self._boundary_conditions['slope']
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+        return self._boundary_conditions['deflection']
+
+
+    def _load_as_SingularityFunction(self, load):
+        x = Symbol('x')
+        if isinstance(load, PointLoad):
+            if load.moment:
+                return load.value*SingularityFunction(x, load.location, S(-2))
+            return load.value*SingularityFunction(x, load.location, S(-1))
+        elif isinstance(load, DistributedLoad):
+            return load.value*SingularityFunction(x, load.start, load.order)
+
+    def apply_loads(self, *loads):
+        """
+        Takes PointLoad and DistributedLoad as input. This method would apply
+        the loads, that are passed as arguments, to the beam object. Internally
+        this method would represent the PointLoads and DistributedLoads using
+        Singularity Functions.
+
+        """
+        for load in loads:
+            if isinstance(load, PointLoad) or isinstance(load, DistributedLoad):
+                self._load += self._load_as_SingularityFunction(load)
+            else:
+                raise TypeError("""apply_loads takes either PointLoad or DistributedLoad objects""")
```

Comment:
I feel that Loads are independent of Beams. In real life, Beams and Loads are considered as different objects. So if we make loads a method under beams, i feel that it would be a bit insensible.
And for the time being, there is no method under load classes. But in the future we can add methods under load classes which would help the load to simplify so that it would be easily rewritten in terms of SingularityFunctions. Bucause, for now the user needs to simplify the loads from the problem . We can have more complex loads as inputs at that point of time and internally it would somehow simplify it. The user can then directly input the loads that would be given in a problem. 
How does this sound?